src/Lecture4.hs

```diff
@@ -184,7 +218,19 @@ instance for the 'Stats' type itself.
 -}
 
 instance Semigroup Stats where
-
+  (<>) :: Stats -> Stats -> Stats
+  (<>) (Stats a b c d e f g h i) (Stats a2 b2 c2 d2 e2 f2 g2 h2 i2) =
+    Stats (a <> a2) (b <> b2) (c <> c2) (d <> d2) (c' e e2) (c' f f2) (c' g g2) (c' h h2) (i <> i2)
+-- hmm
+-- just want to pairwise zip the args
+-- could gmappend from Data.Semigroup.Generic work?
+--   (<>) = gmappend
+
+c' :: Semigroup a => Maybe a -> Maybe a -> Maybe a
+c' a Nothing = a
+c' Nothing a2 = a2
+c' (Just !a) (Just !a2) = Just $ a <> a2
+-- doesn't look like I need to make strict the first two pattern matches here (on Nothings).
```

Comment:
Your comment is absolutely correct 👌🏻 
You can write it a bit shorter though by pattern matching only on the final result:

```suggestion
c' a b = case a <> b of
    Nothing -> Nothing
    Just !x -> Just x
```